lib/services/messenger/translators/discourse.ts

```diff
@@ -309,24 +306,18 @@ export class DiscourseTranslator extends TranslatorScaffold implements Translato
 	 * @returns        Promise that resolves to the details required to construct an emitter.
 	 */
 	public messageIntoEmitterConstructor(message: TransmitInformation): Promise<DiscourseConstructor> {
-		// Go looking through all the data hubs we've been provided for a token.
-		const promises: Array<Promise<string>> = _.map(this.hubs, (hub) => {
-			return hub.fetchValue(message.target.username, 'discourse', 'token');
-		});
 		// Generic has `-` at the end, Discourse has `_` at the beginning
 		const convertedUsername = /-$/.test(message.target.username)
 			? `_${message.target.username.replace(/-$/, '')}`
 			: message.target.username
 		;
-		return Promise.any(promises)
-		.then((token) => {
-			// Pass back details that may be used to connect.
-			return {
-				token,
-				username: convertedUsername,
-				instance: this.connectionDetails.instance,
-				type: ServiceType.Emitter,
-			};
+		// Pass back details that may be used to connect.
+		return Promise.resolve({
```

Comment:
AFAICT This doesn't need return a promise anymore, since the only place this is used is here https://github.com/resin-io-modules/resin-procbots/blob/2c928043584e70dafc35fee0a8932837e5051c8a/lib/services/messenger.ts#L86 it can just return an object.